### PR TITLE
planner: update the scan-row-size calculation formula in model2

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #35240

Problem Summary: planner: update the scan-row-size calculation formula in model2

### What is changed and how it works?

see https://github.com/pingcap/tidb/issues/38967

planner: update the scan-row-size calculation formula in model2.

Current formulas:
1. index-scan-row-size = row-size(all-index-cols);
2. table-scan-row-size = row-size(all-table-cols);
3. tiflash-scan-row-size = row-size(required-cols);

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/testdata/enforce_mpp_suite_out.json [hunk 1]
```diff
@@ -31,30 +31,30 @@
       {
         "SQL": "explain format='verbose' select count(*) from t where a=1",
         "Plan": [
-          "StreamAgg_24 1.00 217.09 root  funcs:count(Column#6)->Column#4",
```
Comment (qw4990) [inline planner/core/testdata/enforce_mpp_suite_out.json:34]: Expected, just update these cost values.

File: planner/core/testdata/integration_suite_out.json [hunk 2]
```diff
@@ -1861,8 +1861,8 @@
       {
         "SQL": "select f, g from t1 where f = 2 and g > 3",
         "Plan": [
-          "IndexReader_6 33.33 752.77 root  index:IndexRangeScan_5",
```
Comment (qw4990) [inline planner/core/testdata/integration_suite_out.json:1864]: Expected, just update these cost values.

File: planner/core/testdata/integration_suite_out.json [hunk 3]
```diff
@@ -2486,65 +2486,65 @@
       {
         "SQL": "explain format = 'verbose' select count(*) from t3",
         "Plan": [
-          "StreamAgg_20 1.00 99.31 root  funcs:count(Column#9)->Column#4",
```
Comment (qw4990) [inline planner/core/testdata/integration_suite_out.json:2489]: Expected, IndexScan is better than TableScan.

File: planner/core/testdata/join_reorder_suite_out.json [hunk 4]
```diff
@@ -6333,8 +6333,8 @@
         "SQL": "select /*+ straight_join() */ * from t1 join t2 on t1.a=t2.a where t1.a not in (select t3.a from t3)",
         "Plan": [
           "HashJoin 9990.00 root  CARTESIAN anti semi join, other cond:eq(test.t1.a, test.t3.a)",
-          "├─TableReader(Build) 3.00 root  data:TableFullScan",
```
Comment (qw4990) [inline planner/core/testdata/join_reorder_suite_out.json:6336]: Expected, IndexScan is better than TableScan.

**Review by time-and-fate — APPROVED** [review]

**Review by Yisaer — APPROVED** [review]

**Pull request merged**